src/display/editor/ink.js

```diff
@@ -268,6 +268,22 @@ class InkEditor extends AnnotationEditor {
     super.remove();
   }
 
+  setParent(parent) {
+    if (!this.parent && parent) {
+      this._uiManager.removeScalingListener(this);
+    } else if (this.parent && parent === null) {
+      this._uiManager.addScalingListener(this);
+    }
```

Comment:
In the second case: the `ink` editor is removed from the view but it can be resized thanks to a zoom and when it's serialized we use the global scale factor:
https://github.com/mozilla/pdf.js/blob/c63906305055d27380ba2f94099a9fe9bb4c1017/src/display/editor/ink.js#L1081
And in the first case, the editor will be visible and will react to the scale changes (through the `ResizeObserver`).

Reply:
Maybe add a short comment describing those different cases?
The `ResizeObserver`-case wasn't immediately obvious, at least to me, when looking at this in isolation.

Reply:
Yep I was doing that.